tests/unit/anchore_engine/clients/test_grype_wrapper.py

```diff
@@ -204,21 +222,71 @@ def test_update_grype_db(grype_db_parent_dir, old_grype_db_dir, grype_db_archive
     assert not os.path.exists(old_grype_db_dir)
 
 
-# TODO This needs to move to a functional test to pass on the ci, as the container with grype
-# is not built before that. Disabling for now. Works locally if you have grype installed.
+def test_get_current_grype_db_metadata(grype_db_dir):
+    # Setup test input
+    grype_wrapper.grype_db_dir = grype_db_dir
```

Comment:
this is exactly the kind of thing that globals will cause. Now this test has effectively tainted the global variable, if this test were to fail, the next test that relies on this value could potentially fail. Figuring out which test did what and when to the global is time-consuming to say the least and all of this can be avoided by removing the global altogether

Reply:
@dspalmer99 do you think the globals could be refactored to a singleton? Or class variables? I really dislike the use of globals here.

Reply:
Maybe, but as I said in another comment, I'd rather do something like make the whole module a singleton class in that case. Which I am not opposed to doing (and would probably prefer) but I'd want to discuss it with the group in our sync first, and in any case it is outside the scope of this pr.